File: problems/12.js

var sum = 0;
var number = 1;
while(sum+=number) {
	number+=1;

	if(divisors(sum) > 500) {
		console.log(sum);
		break;
	}
}

function divisors (number) {
	var div = 0;
	
	for(var i=0; i<=number; i++) {
		if(number%i === 0) div++;
	}
	console.log(div);
	return div;
}

/* 	this is waaaayyy too slow 
	found something like 380 divisor in 15min
*/

File: problems/12.py
sum = 0
number = 1

def divisors (number):
	div = 0
	
	for i in range (1, (number + 1)):
		if number%i == 0:
			div+=1

	return div

while True:
	sum+=number
	number+=1

	if divisors(sum) > 500:
		print sum
		break